src/main.ts

```diff
@@ -65,13 +77,91 @@ function getContext(): any {
 /**
  * Gets the value stored under the reporter key.
  */
+// TODO: The following ts-ignore comment has been added to temporarily suppress compilation errors
+// as the function is not currently used. Remove this when this function has been used somewhere.
+// @ts-ignore
 function getReporter(): any {
   return STORE.reporters;
 }
 
+/**
+ * Registers and stores actions.
+ * @param action an action to register
+ */
+export function registerAction(action: any): void {
+  if (!ACTIONS.includes(action)) {
+    ACTIONS.push(action);
+  }
+}
+
+/**
+ * Executes an array of specs using the current context.
+ * @param specs an array of specs.
+ */
+export async function runSpecs(specs: spec[]) {
+  for (var spec of specs) {
+    if (!ACTIONS.includes(spec.action)) {
+      continue;
+    }
+    let context = getContext();
+    let action = new spec.action();
+    let args = getVariables(spec.args);
+    let title = spec.title;
+    let outs = spec.outs;
+    let results = await action.execute(title, args, outs, context);
```

Comment:
We can catch errors here using `try-catch`

Reply:
Done